File: solution_zzj/34._Find_First_and_Last_Position_of_Element_in_Sorted_Array.cc
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> ans = {-1, -1};
        int l = 0, r = nums.size() - 1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (nums[mid] == target) {
                ans[0] = mid, ans[1] = mid;
                int i = mid, j = mid;
                while (i-1>=0 && nums[i-1]==target) {
                    ans[0] = i-1;
                    i--;
                }
                while (j+1<nums.size() && nums[j+1]==target) {
                    ans[1] = j+1;
                    j++;
                }
                return ans;
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return ans;
    }
};

File: Weijia/34. Find First and Last Position of Element in Sorted Array.py
class Solution(object):
    def searchRange(self, nums, target):
        if len(nums)==0:
            return [-1,-1]
        elif len(nums)==1 and nums[0]==target:
            return [0,0]
        elif len(nums)==1 and nums[0]!=target:
            return [-1,-1]
        else:
            s1=0
            s2=len(nums)
            while s1<s2:
                mid=(s1+s2)//2
                if nums[mid]<target:
                    s1=mid+1
                elif nums[mid]>target:
                    s2=mid-1
                else:
                    if mid+1<len(nums):
                        if nums[mid]==nums[mid+1]:
                            return [mid,mid+1]
                    if mid-1>=0:
                        if nums[mid]==nums[mid-1]:
                            return [mid-1,mid]
                    else:
                        return [mid,mid]
            return [-1,-1]
        
